testharness/src/test/java/bio/terra/policy/service/pao/PaoUpdateTest.java

```diff
@@ -0,0 +1,487 @@
+package bio.terra.policy.service.pao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.policy.common.exception.DirectConflictException;
+import bio.terra.policy.common.exception.InternalTpsErrorException;
+import bio.terra.policy.common.model.PolicyInput;
+import bio.terra.policy.common.model.PolicyInputs;
+import bio.terra.policy.common.model.PolicyName;
+import bio.terra.policy.service.pao.graph.model.PolicyConflict;
+import bio.terra.policy.service.pao.model.Pao;
+import bio.terra.policy.service.pao.model.PaoComponent;
+import bio.terra.policy.service.pao.model.PaoObjectType;
+import bio.terra.policy.service.pao.model.PaoUpdateMode;
+import bio.terra.policy.service.policy.PolicyMutator;
+import bio.terra.policy.service.policy.model.PolicyUpdateResult;
+import bio.terra.policy.testutils.LibraryTestBase;
+import java.util.Collections;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class PaoUpdateTest extends LibraryTestBase {
+  private static final Logger logger = LoggerFactory.getLogger(PaoUpdateTest.class);
+  private static final String TEST_NAMESPACE = "test_namespace";
+  private static final String TEST_FLAG_POLICY_A = "test_flag_a";
+  private static final String TEST_FLAG_POLICY_B = "test_flag_b";
+  private static final String TEST_DATA_POLICY_X = "test_data_x";
+  private static final String TEST_DATA_POLICY_Y = "test_data_y";
+  private static final String DATA_KEY = "key";
+  private static final String DATA1 = "data1";
+  private static final String DATA2 = "data2";
+  private static final String DATA3 = "data3";
+
+  @Autowired private PaoService paoService;
+
+  @Test
+  void unknownFlagCombinerTest() throws Exception {
```

Comment:
nit: test naming should summarize the expected outcome as well
https://engdoc.corp.google.com/eng/doc/devguide/testing/unit/best-practices.md?cl=head&polyglot=java#naming

Reply:
Ugh! I'll try...